env0/resource_project_policy.go

```diff
@@ -0,0 +1,174 @@
+package env0
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourcePolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePolicyCreate,
+		ReadContext:   resourcePolicyRead,
+		UpdateContext: resourcePolicyUpdate,
+		DeleteContext: resourcePolicyReset,
+
+		Importer: &schema.ResourceImporter{StateContext: resourcePolicyImport},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "id of the policy",
+				Computed:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "id  of the project",
+				Required:    true,
+			},
+			"number_of_environments": {
+				Type:        schema.TypeInt,
+				Description: "number of environments per project, 0 indicates no limit",
+				Optional:    true,
+			},
+			"number_of_environments_total": {
+				Type:        schema.TypeInt,
+				Description: "number of environments total, 0 indicates no limit",
+				Optional:    true,
+			},
```

Comment:
I think we should still be validating that these int are positive (and larger than `0`), otherwise API would fail so that's a bad UX

Reply:
As noted in the field description, 0 is being used as a sentinel value to indicate no limit. Added validation to handle negative numbers.

Reply:
Right, haven't seen it. What would happen if a user had set value `1` and then removed the attribute completely?